python/ray/air/_internal/checkpoint_manager.py

```diff
@@ -382,15 +382,21 @@ def _get_checkpoint_score(
         checkpoint_score_attribute = (
             self._checkpoint_strategy.checkpoint_score_attribute
         )
-        if checkpoint_score_attribute not in checkpoint.metrics:
-            logger.error(
-                f"Result dict has no key: {checkpoint_score_attribute}. "
-                f"checkpoint_score_attr must be set to a key in the "
-                f"result dict. Valid keys are: {list(checkpoint.metrics.keys())}"
-            )
-            checkpoint_result = float("-inf")
+        if checkpoint_score_attribute:
+            try:
+                checkpoint_result = unflattened_lookup(
+                    checkpoint_score_attribute, checkpoint.metrics
+                )
```

Comment:
I don't think this will work for using a metric containing `/`.

Unflattened dict:

```
metrics = {"a": {"b/c": 1.0}}
unflattened_lookup(metrics, "a/b/c") -> metrics["a"]["b"]["c"] not found
```

Reply:
We can just flatten the metrics at this point, then access using the `checkpoint_score_attribute` (and don't keep the flattened copy around).

Reply:
Yes, good idea!